cmsplugin_filer_image/migrations/0002_delete_duplicate_thumbnailoptions.py

```diff
@@ -0,0 +1,33 @@
+from django.db import migrations
+from django.db.models import Count
+
+
+def forward(apps, schema_editor):
+    """
+    Get all the duplicates for ThumbnailOption.
+    Point each FilerImage where the duplicates are used to the correct
+    ThumbnailOption.
+    Delete remaining ThumbnailOption duplicates.
+    """
+    ThumbnailOption = apps.get_model("cmsplugin_filer_image", "ThumbnailOption")
+    unique_fields = ["name", "width", "height", "crop", "upscale"]
+
+    duplicates = ThumbnailOption.objects.values(*unique_fields) \
+        .annotate(count_id=Count('id')) \
+        .filter(count_id__gt=1)
+
+    for duplicate in duplicates:
+        qs = ThumbnailOption.objects.filter(
+            **{x: duplicate[x] for x in unique_fields})
+        id_to_keep = qs[:1][0].id
```

Comment:
Since you need all the elements from the qs you can drop the slicing [:1], [1:] so that the only one query gets executed and not 2.